x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/DelimitedFileStructureFinderFactory.java

```diff
@@ -14,6 +14,7 @@
 
 public class DelimitedFileStructureFinderFactory implements FileStructureFinderFactory {
 
+    static final double DEFAULT_BAD_ROWS_PERCENTAGE = 0.10d;
```

Comment:
I would rename it to sth resembling the `allowedFractionOfBadLines` parameter in the interface, e.g.: `DEFAULT_ALLOWED_FRACTION_OF_BAD_LINES`.
Also I'd consider employing the word "max": `maxAllowedFractionOfBadLines` but I do not feel very strongly about it.

Reply:
I would have two constants here, one for the case where we have been told the exact delimiter and one where we have been told it's delimited but not the exact delimiter.

I also agree that it would be better to use `FRACTION` instead of `PERCENTAGE` as the value is not a percentage.

How about:

```
    static final double DELIMITER_OVERRIDDEN_ALLOWED_FRACTION_OF_BAD_LINES = 0.10;
    static final double FORMAT_OVERRIDDEN_ALLOWED_FRACTION_OF_BAD_LINES = 0.05;
```